alacritty_terminal/src/grid/storage.rs

```diff
@@ -22,10 +9,35 @@ use crate::index::Line;
 /// Maximum number of invisible lines before buffer is resized
 const TRUNCATE_STEP: usize = 100;
 
+/// An optimized ring buffer for fast indexing and rotation
+///
+/// The [`Storage::rotate`] and [`Storage::rotate_up`] functions are fast
+/// modular additions on the internal `zero` field. As compared with
+/// [`slice::rotate_left`] which must rearrange items in memory.
+///
+/// As a consequence, both [`Index`] and [`IndexMut`] are reimplemented for this
+/// type to account for the zeroth element not always being at the start of the
+/// allocation.
+///
+/// Because certain [`Vec`] operations are no longer valid on this type, no
+/// [`Deref`](std::ops::Deref) implementation is provided. Anything from `Vec`
+/// that should be exposed must be done so manually.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Storage<T> {
     inner: Vec<Row<T>>,
+
+    /// Starting point for the storage of rows
+    ///
+    /// This value represents the starting line offset within the ring buffer. Doing things this
+    /// way allows for scrolling stored lines in a performant way.
```

Comment:
> Doing things this way allows for scrolling stored lines in a performant way.

We don't need to reiterate these things on the documentation of a field. This kind of documentation should be on the struct or module (I think it's already sufficiently covered in the struct docs).